### PR TITLE
Add support for Docker HostConfig SecurityOpt through che.env

### PR DESCRIPTION
**Pull request by hkolvenbach**
Signed-off-by: Hanno Kolvenbach <kolvenbach@silexica.com>

### What does this PR do?
Port SecurityOpt Pull Request to Che6 (https://github.com/eclipse/che/pull/6856)

### What issues does this PR fix or reference?
See https://github.com/eclipse/che/pull/6856

#### Release Notes

- Add new config variable che.docker.securityopt / CHE_DOCKER_SECURITYOPT
- Support for GDB in unprivileged docker containers

#### Docs PR

https://github.com/eclipse/che-docs/pull/305

### CODE REVIEW

File: infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/model/DockerContainerConfig.java [hunk 1]
```diff
@@ -546,6 +566,7 @@ public int hashCode() {
         getCpuSet(),
         getDependsOn(),
         getDns(),
+        getSecurityOpt(),
```
Comment (garagatyi) [inline infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/model/DockerContainerConfig.java:569]: Can you also modify toString method?

File: infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/provisioner/securityopt/SecurityOptProvisioner.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  @Inject
+  public SecurityOptProvisioner(SecurityOptProvider securityOptProvider) {
+    this.securityOptProvider = securityOptProvider;
+  }
+
+  @Override
+  public void provision(DockerEnvironment internalEnv) throws InfrastructureException {
+    for (DockerContainerConfig containerConfig : internalEnv.getContainers().values()) {
+      // TODO
```
Comment (garagatyi) [inline infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/provisioner/securityopt/SecurityOptProvisioner.java]: Can you remove todo?

**Review by garagatyi — APPROVED** [review]
Looks good!

File: infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/model/DockerContainerConfig.java (hunk 1 above)
Comment (hkolvenbach) [reply infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/model/DockerContainerConfig.java:569]: yes, overlooked that. done!

File: infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/provisioner/securityopt/SecurityOptProvisioner.java (hunk 2 above)
Comment (hkolvenbach) [reply infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/provisioner/securityopt/SecurityOptProvisioner.java]: done

**Review by riuvshin — APPROVED** [review]

**Review by skabashnyuk — APPROVED** [review]

**Comment by garagatyi** [comment]
@benoitf can you take a look?

**Review by benoitf — APPROVED** [review]

**Comment by benoitf** [comment]
ci-build

**Pull request merged**